ffi/src/repr_c.rs

```diff
@@ -117,6 +206,40 @@ impl<'itm, R: CTypeConvert<'itm, S, C> + Clone, S: Cloned, C: ReprC>
     }
 }
 
+impl<'itm, R: NonLocal<S>, S: Cloned> CWrapperOutput<&'itm S> for &'itm R {
+    type ReturnType = LocalRef<'itm, R>;
+}
+impl<R: NonLocal<S>, S: Cloned> COutPtr<&S> for &R {
+    type OutPtr = R::ReprC;
+}
+impl<'itm, R: NonLocal<S> + CTypeConvert<'itm, S, R::ReprC> + Clone, S: Cloned + 'itm>
+    COutPtrWrite<&'itm S> for &'itm R
+{
+    unsafe fn write_out(self, out_ptr: *mut R::ReprC) {
+        let mut store = <(Option<R::ReprC>, R::RustStore)>::default();
+        // NOTE: Bypasses the erroneous lifetime check.
+        // Correct as long as `R::into_repr_c` doesn't return a reference to the store (`R: NonLocal`)
+        let store_borrow = &mut *addr_of_mut!(store);
+        CTypeConvert::<&S, _>::into_repr_c(self, store_borrow);
+
+        // NOTE: None value indicates a bug in the implementation
+        #[allow(clippy::expect_used)]
+        out_ptr.write(store.0.expect("Store must be initialized"));
+    }
+}
+impl<'itm, R: NonLocal<S> + CTypeConvert<'itm, S, R::ReprC> + Clone + 'itm, S: Cloned + 'itm>
+    COutPtrRead<&'itm S> for LocalRef<'itm, R>
+{
+    unsafe fn try_read_out(out_ptr: Self::OutPtr) -> Result<Self> {
+        let mut store = Default::default();
+        // NOTE: Bypasses the erroneous lifetime check.
+        // Correct as long as `R::into_repr_c` doesn't return a reference to the store (`R: NonLocal`)
```

Comment:
Maybe? (in other places too)

```suggestion
        // Correct as long as `R::try_from_repr_c` doesn't return a reference to the store (`R: NonLocal`)
```